src/gui/folder.cpp

```diff
@@ -108,15 +110,62 @@ Folder::Folder(const FolderDefinition &definition,
     _scheduleSelfTimer.setInterval(SyncEngine::minimumFileAgeForUpload);
     connect(&_scheduleSelfTimer, &QTimer::timeout,
         this, &Folder::slotScheduleThisFolder);
+
+    //initialize wamp connection
+    QWamp::Session *session;
+    _webSocket.reset(new QTcpSocket());
+//    QByteArray wsUrl = qgetenv("OWNCLOUD_WEBSOCKET_URL");
+    QByteArray wsUrl = "ws://localhost:8080/ws";
+    QObject::connect(_webSocket.data(), &QTcpSocket::connected, [&]() {
+
+        QString sessionName("sessionName");
+        session = new QWamp::Session(sessionName, *_webSocket.data(), QWamp::Session::MessageFormat::Msgpack, true);
+
+        QObject::connect(session, &QWamp::Session::joined, [&](qint64 s) {
+          qDebug() << "Session joined to realm1 with session ID " << s;
+
+          session->subscribe("etag-changed-channel", [&](const QVariantList& args, const QVariantMap& options){
+              qDebug() << "Event received";
+          });
+        });
+
+        QObject::connect(session, &QWamp::Session::started, [&]() {
+          session->join("realm1");
+        });
+        session->start();
+      });
+
+    QObject::connect(_webSocket.data(), QOverload<QAbstractSocket::SocketError>::of(&QAbstractSocket::error), [&](QAbstractSocket::SocketError err){
```

Comment:
QOverload requires Qt 5.7    (before that, we'd have to use static_cast)

(Actually, i've been thinking about copying QOverload within some header so we can use it while still being compatible with Qt 5.6)

Reply:
well - what ever works - tell me whyt to do